integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/highlight/HighlighterFastVectorIT.java

```diff
@@ -0,0 +1,222 @@
+								" style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft <em>rock</em>."
+						)
+				);
+	}
+
+	@Test
+	public void boundaryScannerSentence() {
+		assumeTrue( "Ignoring the test on purpose as the results look not ok.", false );
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<String>> highlights = scope.query().select(
+						f -> f.highlight( "string" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "rock" ) )
+				.highlighter( h -> h.highlighterFastVector()
+						.boundaryScanner()
+						.sentence()
+						.locale( Locale.ENGLISH )
+				)
+				.toQuery();
+
+		assertThatHits( highlights.fetchAllHits() )
+				.hasHitsAnyOrder(
+						// this test will fail at lest on ES as it adds one more unexpected sentence to the first match, while I'd expect to have single sentence per line.
+						// produces :
+						//		"Scorpions are a German <em>rock</em> band formed in Hanover in 1965 by guitarist Rudolf Schenker. Since the band's inception, its musical style has ranged from hard rock, heavy metal and glam metal to soft rock. ",
+						//		"Since the band's inception, its musical style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft <em>rock</em>.
+						Arrays.asList(
+								"Scorpions are a German <em>rock</em> band formed in Hanover in 1965 by guitarist Rudolf Schenker.",
+								"Since the band's inception, its musical style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft <em>rock</em>."
+						)
+				);
+	}
+
+	@Test
+	public void styledSchema() {
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<String>> highlights = scope.query().select(
+						f -> f.highlight( "string" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "another" ) )
+				.highlighter( h2 -> h2.highlighterFastVector().tagSchema( HighlighterTagSchema.STYLED ) )
+				.toQuery();
+
+		assertThatHits( highlights.fetchAllHits() )
+				.hasHitsAnyOrder(
+						Arrays.asList( "some <em class=\"hlt1\">another</em> value" ),
+						Arrays.asList( "some yet <em class=\"hlt1\">another</em> value" )
+				);
+	}
+
+	@Test
+	public void boundaryCharacters() {
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<String>> highlights = scope.query().select(
+						f -> f.highlight( "string" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "time" ) )
+				.highlighter( h2 -> h2.highlighterFastVector()
+						.boundaryScanner()
+						.chars()
+						.locale( Locale.ENGLISH )
+						.boundaryChars( "-" )
+						.boundaryMaxScan( 5 )
+				)
+				.toQuery();
+
+		assertThatHits( highlights.fetchAllHits() )
+				.hasHitsAnyOrder(
+						// some unexplainable result >_<
+						Arrays.asList( "at has - dash in - it from <em>time</em> - to -some useless text in between - <em>time</em> to see - how - boundary_chars " )
```

Comment:
What do you expect instead?

Reply:
Good question 😃 !
Since the boundary scanner is supposed to define how to break the highlighted fragments... I'd expect that if the boundary characters are set to `-` then the fragments would've been cut at `-` rather than cutting the words in the middle... 
And even with the default fragment size of 100 .. this result is still different from the one that was using the sentence scanner - there was a full second sentence but it wasn't highlighted. 
these settings work exactly the same in ES and in what we have for Lucene, but if you'd ask me to explain why it behaves as it is ... 🥲